module/zfs/vdev_queue.c

```diff
@@ -171,6 +171,24 @@ uint32_t zfs_vdev_rebuild_max_active = 3;
 int zfs_vdev_async_write_active_min_dirty_percent = 30;
 int zfs_vdev_async_write_active_max_dirty_percent = 60;
 
+/*
+ * To reduce effects of non-interactive I/O on interactive I/O latency
+ * the first are limited to *_min_active while there are second active,
+ * plus at least this number of I/Os after in case interactive return.
+ */
+int zfs_vdev_nia_delay = 5;
+
+/*
+ * Some HDDs tend to prioritize sequential I/O so high, that concurrent
+ * random I/O latency reaches several seconds.  On some HDDs it happens
+ * even if sequential I/Os are submitted one at a time, and so setting
+ * *_max_active to 1 does not help.  To handle this in case of scrub
```

Comment:
How about:

```
 * ...
 * *_max_active to 1 does not help.  To prevent non-interactive I/Os, like scrub, from
 * monopolizing the device no more than zfs_vdev_nia_credit I/Os can be sent while
 * there are outstanding incomplete interactive I/Os.  This enforced wait ensures the
 * HDD services the interactive I/O within a reasonable amount of time.
 */
```
device
are outstanding

Reply:
Thanks.